plugins/org.locationtech.udig.render.wms.basic/src/org/locationtech/udig/render/wms/basic/internal/WMSRequestUtil.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * uDig - User Friendly Desktop Internet GIS client
+ * https://locationtech.org/projects/technology.udig
+ * (C) 2021, Eclipse Foundation
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * (http://www.eclipse.org/legal/epl-v10.html), and the Eclipse Distribution
+ * License v1.0 (http://www.eclipse.org/org/documents/edl-v10.html).
+ */
+package org.locationtech.udig.render.wms.basic.internal;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.geotools.ows.wms.WebMapServer;
+
+/**
+ * @author fgdrf
+ *
+ */
+public class WMSRequestUtil {
+
+    public static String getRequestImageFormatBestMatch(WebMapServer webMapServer,
+            String[] preferredImageFormats) {
+        List<String> formats = webMapServer.getCapabilities().getRequest().getGetMap().getFormats();
```

Comment:
`getGetMap()` looks odd to me :D we may talk about this at some day?

Reply:
Its the OGC WMS specification implementation (from xsd schema of capabilities xml document to xml beans)